compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/CodeAnalyzer.java

```diff
@@ -2782,6 +2785,12 @@ public void visit(BLangExpressionStmt exprStmtNode) {
         this.checkStatementExecutionValidity(exprStmtNode);
         analyzeExpr(exprStmtNode.expr);
         validateExprStatementExpression(exprStmtNode);
+        if (exprStmtNode.expr.getKind() == NodeKind.INVOCATION) {
+            BLangInvocation invocation = (BLangInvocation) exprStmtNode.expr;
+            if (types.isNeverTypeOrStructureTypeWithARequiredNeverMember(invocation.getBType())) {
+                this.statementReturns = true;
```

Comment:
I think we should use a separate variable like `statementPanics` here, with panics, etc. Let's create an issue for this.